examples/sheep_wolf.jl

```diff
@@ -0,0 +1,164 @@
+        wolf = Wolf(id, (0,0), energy, wolf_reproduce, Δenergy_wolf)
+        add_agent!(wolf, model)
+    end
+    for n in nodes(model)
+        id += 1
+        fully_grown = rand(Bool)
+        fully_grown ? countdown = regrowth_time : countdown = rand(1:regrowth_time) - 1
+        grass = Grass(id, (0,0), fully_grown, regrowth_time, countdown)
+        add_agent!(grass, n, model)
+    end
+    return model
+end
+
+function move!(agent, model)
+    neighbors = node_neighbors(agent, model)
+    cell = rand(neighbors)
+    move_agent!(agent, cell, model)
+end
+
+function eat!(wolf::Wolf, sheep, model)
+    if !isempty(sheep)
+        dinner = rand(sheep)
+        kill_agent!(dinner, model)
+        wolf.energy += wolf.Δenergy
+    end
+end
+
+function eat!(sheep::Sheep, grass_array, model)
+    isempty(grass_array) ? (return nothing) : nothing
+    grass = grass_array[1]
+    if grass.fully_grown
+        sheep.energy += sheep.Δenergy
+        grass.fully_grown = false
+        grass.countdown = grass.regrowth_time
+    end
+end
+
+function agent_step!(sheep::Sheep, model)
+    move!(sheep, model)
+    sheep.energy -= 1
+    agents = get_node_agents(sheep.pos, model)
+    dinner = filter!(x->isa(x, Grass), agents)
+    eat!(sheep, dinner, model)
+    if sheep.energy <= 0
+        kill_agent!(sheep, model)
+        return nothing
+    end
+    if rand() <= sheep.reproduction_prob
+        reproduce!(sheep, model)
+    end
+end
+
+function agent_step!(wolf::Wolf, model)
+    move!(wolf, model)
+    wolf.energy -= 1
+    agents = get_node_agents(wolf.pos, model)
+    dinner = filter!(x->isa(x, Sheep), agents)
+    eat!(wolf, dinner, model)
+    if wolf.energy <= 0
+        kill_agent!(wolf, model)
+        return nothing
+    end
+    if rand() <= wolf.reproduction_prob
+        reproduce!(wolf, model)
+        return nothing
+    end
+end
+
+function agent_step!(grass::Grass, model)
+    if !grass.fully_grown
+        if grass.countdown <= 0
+            grass.fully_grown = true
+            grass.countdown = grass.regrowth_time
+        else
+            grass.countdown -= 1
+        end
+    end
+end
+
+function reproduce!(agent, model)
+    agent.energy /= 2
+    id = nextid(model)
+    energy = rand(1:agent.Δenergy*2) - 1
+    A = typeof(agent)
+    spawn = A(id, (0,0), energy, agent.reproduction_prob, agent.Δenergy)
+    add_agent!(spawn, model)
+    return nothing
+end
+
+function model_step!(model, df)
+    agents = values(model.agents)
+    model.properties[:step] += 1
+    step = model.properties[:step]
+    n_sheep = count(x->typeof(x) == Sheep, agents)
+    n_wolves = count(x->typeof(x) == Wolf, agents)
+    n_grass = count(x->(typeof(x) == Grass) && x.fully_grown, agents)
+    push!(df, [step n_sheep n_wolves n_grass])
+end
+
+
+n_steps = 500
+Random.seed!(23182)
+model = initialize_model()
+results = DataFrame(step=Int[], n_sheep=Int[], n_wolves=Int[], n_grass=Int[])
+step!(model, agent_step!, x->model_step!(x, results), n_steps)
```

Comment:
This is not needed anymore with the new data collection system. You can replace this highlighted section with the following:

```julia
n_steps = 500
Random.seed!(23182)
model = initialize_model()
sheep(a) = typeof(a) == Sheep
wolves(a) = typeof(a) == Wolf
grass(a) = typeof(a) == Grass && a.fully_grown
adata = [(sheep, count), (wolves, count), (grass, count)]             
results, _ = run!(model, agent_step!, n_steps; adata = adata)
```

Reply:
Thank you. I pushed the requested changes. 

I was wondering if there is a reason for using `function(variable)` as the default naming scheme? Unfortunately, this requires wrapping the variable in Symbol and cols in order to plot with StatsPlots: `cols(Symbol("count(sheep)"))`. It seems like `:function_variable` would be more concise.

Reply:
There's also [aggname](https://juliadynamics.github.io/Agents.jl/dev/api/#Agents.aggname) to help with this.

Exactly why we're using this convention&mdash;I'm unsure. It's probably a relic from the old data collection methods. We could have a discussion about that.

Reply:
In that case, I'll open a separate issue.